db/migrate/20200410152726_create_machine_snacks.rb

```diff
@@ -0,0 +1,9 @@
+class CreateMachineSnacks < ActiveRecord::Migration[5.1]
+  def change
+    create_table :machine_snacks do |t|
+      t.references :machine, foreign_key: true
+      t.references :snack, foreign_key: true
```

Comment:
this was a typo that delayed me a bit. when i created this migration i called  
`snacks:references` instead of `snack:references` so this line read  
`t.references :snacks, foreign_key: true` and was throwing errors like nobody's business

Reply:
Glad you were able to catch that.. this is a frequent thing that I've done before as well.